CMake: add install rules, respect BUILD_EXAMPLES option

foolnotion opened the pull request:
Hi,

I hope this will be useful, it makes it easier to package and utilize the library. This PR attempts to add some install rules in `CMakeLists.txt` and additionally checks the `BUILD_EXAMPLES` variable to see if the example executables are needed.

I am not super expert in CMake but these changes work for me. If you prefer to integrate the install rules differently (e.g. include a separate file like `cmake/install-rules.cmake`) please feel free to request any changes.

Thanks

jmuehlig commented:
Thanks a lot, I appreciate your contribution! I suggest two slight modifications before merging your PR into the next version:

- Could you move the custom target `perf-list` so that it is available even if `BUILD_EXAMPLES` is not set, enabling the user to generate counter lists?
- Please add documentation as a new section to the documentation (`docs/build.md`, after the "Using CMake and ExternalProject" section).

foolnotion commented:
Hi, sorry it's been so slow, other stuff got in the way. I've made further changes to `CMakeLists.txt` to make it (IMHO) more robust, notably I've replaced `ExternalProject_Add` with `FetchContent_Declare`, which seems to work very well (as a side note, using an established tool like [cmake-init](https://github.com/friendlyanon/cmake-init) to make the project CMake-friendly would probably be even better). But please do a test  merge and try this out since I've had to use my own fork as an URL.

I've also added an install rule for all the example programs, but I wasn't sure if `perf-list` needs anything to be installed.

Thanks,
